lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_merge_strings_files.rb

```diff
@@ -0,0 +1,75 @@
+module Fastlane
+  module Actions
+    class IosMergeStringsFilesAction < Action
+      def self.run(params)
+        UI.message "Merging strings files: #{params[:paths].inspect}"
+
+        duplicates = Fastlane::Helper::Ios::L10nHelper.merge_strings(paths: params[:paths], output_path: params[:destination])
+        duplicates.each do |dup_key|
+          UI.important "Duplicate key found while merging the `.strings` files: `#{dup_key}`"
+        end
+        duplicates
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Merge multiple `.strings` files into one'
+      end
+
+      def self.details
+        <<~DETAILS
+          Merge multiple `.strings` files into one.
+
+          Especially useful to prepare a single `.strings` file merging strings from both `Localizable.strings` from
+          the app code — typically previously extracted from `ios_generate_strings_file_from_code` —
+          and string files like `InfoPlist.strings` — which values may not be generated from the codebase but
+          manually maintained by developers.
+
+          The action only supports merging files which are in the OpenStep (`"key" = "value";`) text format (which is
+          the most common format for `.strings` files, and the one generated by `genstrings`), but can handle the case
+          of different files using different encodings (UTF8 vs UTF16) and is able to detect and report duplicates.
+          It does not handle strings files in XML or binary-plist formats `.strings` files (which are valid but more rare)
+        DETAILS
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :paths,
+            env_name: 'FL_IOS_MERGE_STRINGS_FILES_PATHS',
+            description: 'The paths of all the `.strings` files to merge together',
+            type: Array,
+            optional: false
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :destination,
+            env_name: 'FL_IOS_MERGE_STRINGS_FILES_DESTINATION',
+            description: 'The path of the merged `.strings` file to generate. If nil, the merge will happen in-place in the first file in the `paths:` list',
+            type: String,
+            optional: true,
+            default_value: nil
```

Comment:
How do you feel about making this parameter required, so that we don't have to specify, implement, and manage the fallback logic to the in-place merge?

Reply:
I actually implemented the in-place logic because I think that's how we'll likely use it in the `Fastfile`, merging `InfoPlist.strings` and `Sites.strings` directly into the `Localizable.strings` immediately after we generated it using `generate_strings_file_from_code`?

Reply:
And to complete that answer, just wanted to note that even if we made this parameter mandatory, we would still want to be sure for the specs to cover the case where the output file would be one of files already listed in the array of input files.

I'm saying this because:
 - This will still likely be our typical use case at call site in our `Fastfiles` as mentioned above
 - But I've so often been bitten in the past by the case of running `cat foo.txt bar.txt >foo.txt` in bash and ending up with a totally empty `foo.txt` file in the end when `foo.txt` is both used as input and (redirect-)output of `cat`.

Given this action is a bit like a smart `cat` in a way (which adds support for files with different encodings and detecting duplicates), I wanted to make sure we didn't risk falling into the same trap.

That's also one of the main reason(†) why the actual implementation [uses an intermediate file](https://github.com/wordpress-mobile/release-toolkit/pull/329/files#diff-6867f769ac6316c7b553d9e5ac93e7aa55d4f9b82bb867e2dc552c8500dd1dacR52) and [copies it back to the destination only once done](https://github.com/wordpress-mobile/release-toolkit/pull/329/files#diff-6867f769ac6316c7b553d9e5ac93e7aa55d4f9b82bb867e2dc552c8500dd1dacR71). Which is why I wanted to be sure to cover that case and test that this specific edge case still worked as expected and was accounted for and made clear in the API that this was supported 🙃 

(†) another reason is for atomic writes, e.g. to not mess up the output file in case something crashes before we finish generating the whole content, similar to how `[write(toFile:atomically:)` does it in Foundation](https://developer.apple.com/documentation/foundation/nsdata/1408033-write)` 😛

Reply:
Sounds good 👍